Update documentation with Dynatrace API v2 usage information

pirgeo opened the pull request:
This PR contains updated documentation for the Dynatrace metrics API v2.
- Issue: https://github.com/micrometer-metrics/micrometer/issues/2295
- Pull request:https://github.com/micrometer-metrics/micrometer/pull/2581

shakuzen commented:
Thank you for the thorough pull request. We don't want to update the site until the corresponding code is released, so I've marked this as `blocked` and we will revisit it when v2 support is added.

jonatan-ivanov commented on src/docs/implementations/dynatrace.adoc:
```diff
@@ -4,25 +4,190 @@ Jon Schneider <jschneider@pivotal.io>
+MeterRegistry registry = new DynatraceMeterRegistry(dynatraceConfig, Clock.SYSTEM);
+----
+
+If you are working with Spring Boot, you can instead put the following properties in your `application-dynatrace.yml` in the `resources` folder.
```
I'm not sure `application-dynatrace.yml` is the right way to do this, I would just say add the right properties:
>DynatraceConfig is an interface with a set of default methods. If, in the implementation of get(String k), rather than returning null, you instead bind it to a property source, you can override the default configuration. For example, Micrometer’s Spring Boot support binds properties prefixed with `management.metrics.export.dynatrace` directly to the DynatraceConfig:

```
management.metrics.export.dynatrace:
    api-version: v2 # Which API do you want to use V1 or V2. (Default: V1)
```

jonatan-ivanov commented on src/docs/implementations/dynatrace.adoc:
```diff
@@ -4,25 +4,190 @@ Jon Schneider <jschneider@pivotal.io>
+
     @Override
     public String apiToken() {
+        // It is *not* recommended to place secrets in config files but rather to read them from env vars, for example.
```
I would just say "read them from a secure source". Environment variables are not necessarily secure/best solution, though they can be ok in certain situations.
There are other occurrences in this doc, could you please fix those too?

jonatan-ivanov commented on src/docs/implementations/dynatrace.adoc:
```diff
@@ -4,25 +4,190 @@ Jon Schneider <jschneider@pivotal.io>
+[source,yml]
+----
+management.metrics.export.dynatrace:
+    enabled: true
```
I don't think `enabled ` is needed. Could you please fix it in other places too?

pirgeo replied:
@jonatan-ivanov From my tests with the Micrometer Dynatrace sample application it seems as if this is needed. If I don't specify it, nothing will be exported. Am I missing a configuration option somewhere else?

shakuzen replied:
If you are referring to the DynatraceSample in the Micrometer repository, it needs to be enabled because the uber sample app contains all of the registries and hence we disable them all by default so you can run with just one registry enabled. This is mostly unique to the structure of the all-in-one sample application and not typical for a user.

pirgeo replied:
That's exactly what I'm referring to. I will remove them then - Thanks!

jonatan-ivanov commented on src/docs/implementations/dynatrace.adoc:
```diff
@@ -4,25 +4,190 @@ Jon Schneider <jschneider@pivotal.io>
+
+*Configuration with URI and API token*
+
+If no local OneAgent is running on the host or the metrics should be sent to a different endpoint (e.g., a different tenant), the Dynatrace v2 exporter can be configured with an explicit endpoint URI and an https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/[API token].
```
`e.g.:` instead of `e.g.,`

jonatan-ivanov commented:
@pirgeo Thank you very much for doing such a comprehensive job (implementing the feature, adding docs and Spring Boot support).

pirgeo commented:
Since 1.8 is released containing code described here, could you take another look? I rebased this PR and updated it to reflect the latest changes in the Dynatrace registry. Thanks!

jonatan-ivanov commented:
@pirgeo Merged, thank you very much for your contributions.
